airflow/utils/context.py

```diff
@@ -245,28 +250,47 @@ class InletEventsAccessors(Mapping[str, LazyDatasetEventSelectSequence]):
 
     _inlets: list[Any]
     _datasets: dict[str, Dataset]
+    _dataset_aliases: dict[str, DatasetAlias]
     _session: Session
 
     def __init__(self, inlets: list, *, session: Session) -> None:
         self._inlets = inlets
-        self._datasets = {inlet.uri: inlet for inlet in inlets if isinstance(inlet, Dataset)}
         self._session = session
+        self._datasets = {}
+        self._dataset_aliases = {}
+
+        for inlet in inlets:
+            if isinstance(inlet, Dataset):
+                self._datasets[inlet.uri] = inlet
+            elif isinstance(inlet, DatasetAlias):
+                self._dataset_aliases[inlet.name] = inlet
```

Comment:
This makes me wonder if we should have something in BaseDataset to encapsulate getting a key out of the classes. That’s refactoring for another day, I guess.

Reply:
Yep, that's a great idea, and I agree that it would be a refactoring in another PR.